pkg/operator/defaultscccontroller/defaultscccontroller.go

```diff
@@ -0,0 +1,232 @@
+package defaultscccontroller
+
+import (
+	"fmt"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog"
+
+	securityv1 "github.com/openshift/api/security/v1"
+	securityv1client "github.com/openshift/client-go/security/clientset/versioned/typed/security/v1"
+	securityv1listers "github.com/openshift/client-go/security/listers/security/v1"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+const (
+	ControllerName = "default-scc-upgradeable"
+)
+
+type Options struct {
+	Config         *rest.Config
+	OperatorClient v1helpers.OperatorClient
+	Recorder       events.Recorder
+	ResyncPeriod   time.Duration
+}
+
+func NewDefaultSCCController(options *Options) (controller *DefaultSCCController, err error) {
+	client, err := securityv1client.NewForConfig(options.Config)
+	if err != nil {
+		err = fmt.Errorf("[%s] failed to create client for security/v1 - %s", ControllerName, err.Error())
+		return
+	}
+
+	defaultSCCSet, err := NewDefaultSCCCache()
+	if err != nil {
+		err = fmt.Errorf("[%s] failed to render default SCC assets - %s", ControllerName, err.Error())
+		return
+	}
+
+	// Create a new SecurityContextConstraints watcher
+	watcher := &cache.ListWatch{
+		ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+			return client.SecurityContextConstraints().List(options)
+		},
+
+		WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+			return client.SecurityContextConstraints().Watch(options)
+		},
+	}
+
+	queue := workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "DefaultSCCController")
+
+	indexer, informer := cache.NewIndexerInformer(watcher, &securityv1.SecurityContextConstraints{}, options.ResyncPeriod, defaultSCCControllerEventHandler(queue), cache.Indexers{})
```

Comment:
use the shared informer factory, not independent informers.

Reply:
done.